cmd/bee-file/main.go

```diff
@@ -0,0 +1,255 @@
+
+	if outDir == "" {
+		outDir = "."
+	} else {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+	}
+	outFilePath := filepath.Join(outDir, metaData.Filename)
+
+	// create output dir if not exist
+	if outDir != "." {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+	}
+
+	// protect any existing file unless explicitly told not to
+	outFileFlags := os.O_CREATE | os.O_WRONLY
+	if outFileForce {
+		outFileFlags |= os.O_TRUNC
+	} else {
+		outFileFlags |= os.O_EXCL
+	}
+
+	// open the file
+	outFile, err := os.OpenFile(outFilePath, outFileFlags, 0o666) // skipcq: GSC-G302
+	if err != nil {
+		return err
+	}
+	defer outFile.Close()
+
+	err = cmdfile.JoinReadAll(j, e.Reference(), outFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// putEntry creates a new file entry with the given reference.
+func putEntry(cmd *cobra.Command, args []string) (err error) {
+	// process the reference to retrieve
+	addr, err := swarm.ParseHexAddress(args[0])
+	if err != nil {
+		return err
+	}
+	// add the fsStore and/or apiStore, depending on flags
+	stores := cmdfile.NewTeeStore()
+	if outDir != "" {
+		err := os.MkdirAll(outDir, 0o777) // skipcq: GSC-G301
+		if err != nil {
+			return err
+		}
+		store := cmdfile.NewFsStore(outDir)
+		stores.Add(store)
+	}
+	if useHttp {
+		store := cmdfile.NewApiStore(host, port, ssl)
+		stores.Add(store)
+	}
+
+	// create metadata object, with defaults for missing values
+	if filename == "" {
+		filename = args[0]
+	}
+	if mimeType == "" {
+		mimeType = defaultMimeType
+	}
+	metadata := entry.NewMetadata(filename)
+	metadata.MimeType = mimeType
+
+	// serialize metadata and send it to splitter
+	metadataBytes, err := json.Marshal(metadata)
+	if err != nil {
+		return err
+	}
+	logger.Debugf("metadata contents: %s", metadataBytes)
+
+	// set up splitter to process the metadata
+	s := splitter.NewSimpleSplitter(stores)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// first add metadata
+	metadataBuf := bytes.NewBuffer(metadataBytes)
+	metadataReader := io.LimitReader(metadataBuf, int64(len(metadataBytes)))
+	metadataReadCloser := ioutil.NopCloser(metadataReader)
+	metadataAddr, err := s.Split(ctx, metadataReadCloser, int64(len(metadataBytes)))
+	if err != nil {
+		return err
+	}
+
+	// create entry from given reference and metadata,
+	// serialize and send to splitter
+	fileEntry := entry.New(addr, metadataAddr)
+	fileEntryBytes, err := fileEntry.MarshalBinary()
+	if err != nil {
+		return err
+	}
+	fileEntryBuf := bytes.NewBuffer(fileEntryBytes)
+	fileEntryReader := io.LimitReader(fileEntryBuf, int64(len(fileEntryBytes)))
+	fileEntryReadCloser := ioutil.NopCloser(fileEntryReader)
+	fileEntryAddr, err := s.Split(ctx, fileEntryReadCloser, int64(len(fileEntryBytes)))
+	if err != nil {
+		return err
+	}
+
+	// output reference to file entry
+	fmt.Println(fileEntryAddr)
```

Comment:
i think the cobra `cmd` has some Println functionality that might be better

Reply:
"better" :D

Reply:
:D

I think that its "better" because if you use the standard `fmt.Println` and the process quits, there's that strange `%` sign showing up when the process exits. I never understood why. The cobra print should mitigate that